doc/syntax.md

```diff
@@ -2,42 +2,323 @@
+end
+
+x integer = 0
+for x in 1 ... 5:
+  <statements>
+end
+```
+
+```
+for <existing_variable_name> in <integer_lower_bound> ... <integer_upper_bound> step <integer_step_size>:
+  <statements>
+end
+
+x integer
+for x in 1 ... 5 step 2:
+  <statements>
+end
+```
+
+```
+for <existing_variable_name> in <real_lower_bound> ... <real_upper_bound> step <real_step_size>:
+  <statements>
+end
+
+x real
+for x in 0.1 ... 0.5 step 0.1:
+  Statements
+end
+```
+
+#### Conditionals
+---------------------------------------------
+NESTML supports the following types of if-else conditionals. The indentation is
+not important. The first example shows if conditional composed of a single if
+block:
+
+```
+if <boolean_expression>:
+  <statements>
+end
+
+if 2 < 3:
+  <statements>
+end
+```
+
+The following example shows an if-else block:
+```
+if <boolean_expression>:
+  <statements>
+else:
+  <statements>
+end
 
-* list of operators
+if 2 < 3:
+  <statements>
+else:
+  <statements>
+end
+```
+The  following examples shows usage of the `elif`-conditionals. After an `if`
+condition an arbitrary number of `elif` conditions can follow. The last condition
+block always concludes with an `end` keyword.
+```
+if <boolean_expression>:
+  <statements>
+elif <boolean_expression>:
+  <statements>
+else:
+  <statements>
+end
+
+if 2 < 3:
+  <statements>
+elif 4>6:
+  <statements>
+else:
+  <statements>
+end
+
+if 2 < 3:
+  <statements>
+elif 4>6:
+
+  <statements>
+end
+```
+
+The last example shows nesting of conditionals.
+```
+if 1 < 4:
+  <statements>
+  if 2 < 3:
+    <statements>
+  end
+end
+```
+### Expressions and operators
+Expressions in NESTML can be specified in a recursive fashion. First, terms are
+valid expressions.
+#### Terms:
+All variables, literals, and function calls are valid terms. Variables are names of
+user defined or predefined variables (`t, e`)
+
+#### List of operators
+For any two valid numeric expressions `a, b`, a boolean expressions `c,c1,c2`, and an
+integer expression `n` the following operators produce valid expressions.
+
+| Operator | Description | Examples |
+|-|-|-|
+|() | Expressions with parentheses | (a) |
+|**|  Power operator. Power is right associative, e.g. a \*\* b \*\* c corresponds a \*\* (b \*\* c) | a \*\* b |
+|+,  -, ~| unary plus, unary minus, bitwise negation | -a, ~c|
+|*, /,%  | Multiplication, Division and Modulo-Operator | a * b, a % b|
+|+, -| Addition and Subtraction |a + b, a - b|
+|<<, >>| & Left and right bit shifts |a << n, a >> n|
+|&, &#124;,^ | Bitwise `and`, `or` and `xor` | a&b, a&#124;b, a~b |
+|<, <=, ==, !=, >=, >|  Comparison operators |a <= b, a != b|
+|not, and, or| & Logical conjunction, disjunction and negation |not c, c1 or c2|
+|?:| The ternary logical operator |c?a:b|
 
 ## Blocks
 
 To structure NESTML files, all content is split into blocks. Blocks
-begin with a keyword specifying the type of the block followed by zero
-or more arguments (depending on the type) and a colon. They are
```

Comment:
It is some pr/merge magic. Comparing the resulting text, I'll find this paragraph there.